Nil pointer error during block fetching

#### System information

Geth version: `v0.5.2-stable`
OS & Version: Windows/Linux/OSX

#### Expected behaviour

Node should be able to fetch new block notified by broadcast from the network.

#### Actual behaviour

It's possible that block fetcher can't find the node and trigger a panic.

#### Steps to reproduce the behaviour

Setup a node and run.

#### Backtrace

````
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x14b5025]

goroutine 1373067 [running]:
github.com/ethereum/go-ethereum/eth.(*handler).fetchHeader(0xc008312e01?, {0xc00038fc40?, 0x1bd3bc0?}, {0x6a, 0x8a, 0x9d, 0xf, 0x42, 0x9b, 0xac, ...}, ...)
        github.com/ethereum/go-ethereum/eth/handler.go:666 +0x25
github.com/ethereum/go-ethereum/beacon/impl/fetcher.(*BlockFetcher).loop.func1.1({0x6a, 0x8a, 0x9d, 0xf, 0x42, 0x9b, 0xac, 0xef, 0xdd, 0xa6, ...})
        github.com/ethereum/go-ethereum/beacon/impl/fetcher/block_fetcher.go:467 +0x93
created by github.com/ethereum/go-ethereum/beacon/impl/fetcher.(*BlockFetcher).loop.func1 in goroutine 1373066
        github.com/ethereum/go-ethereum/beacon/impl/fetcher/block_fetcher.go:464 +0xa5
````

--- a/eth/handler.go
+++ b/eth/handler.go
@@ -663,19 +663,27 @@ func (h *handler) BroadcastBlock(block *types.Block, propagate bool) {
 
 // fetchHeader is the bridge to use a hash to get the header from `eth` protocol.
 func (h *handler) fetchHeader(id string, hash common.Hash, sink chan *eth.Response) (*eth.Request, error) {
-	return h.peers.peer(id).RequestOneHeader(hash, sink)
+	peer := h.peers.peer(id)
+	if peer == nil {
+		return nil, errPeerNotRegistered
+	}
+	return peer.RequestOneHeader(hash, sink)
 }
 
 // fetchBodies is the bridge to use a hash array to get the block bodies from `eth` protocol.
 func (h *handler) fetchBodies(id string, hashes []common.Hash, sink chan *eth.Response) (*eth.Request, error) {
-	return h.peers.peer(id).RequestBodies(hashes, sink)
+	peer := h.peers.peer(id)
+	if peer == nil {
+		return nil, errPeerNotRegistered
+	}
+	return peer.RequestBodies(hashes, sink)
 }
 
 // fetchSidecars is the bridge to use a hash array to get the blob sidecars from `beacon` protocol.
 func (h *handler) fetchSidecars(id string, hashes []common.Hash, sink chan *beaconproto.Response) (*beaconproto.Request, error) {
 	peer := h.peers.beacon(id)
 	if peer == nil {
-		return nil, errors.New("unknown beacon peer")
+		return nil, errPeerNotRegistered
 	}
 	return peer.RequestBatchBlobs(hashes, sink)
 }

